src/EditorFeatures/CSharpTest/ExtractMethod/ExtractMethodTests.cs

```diff
@@ -7373,28 +7373,7 @@ static void Main(string[] args)
         }
     }
 }";
-            var expected = @"class Test
-{
-    void method()
-    {
-        static void Main(string[] args)
-        {
-            int v = 0;
-            for(int i=0 ; i<5; i++)
-            {
-                v = NewMethod(v, i);
```

Comment:
@CyrusNajmabadi To elaborate, my point is that you cannot expect flow analysis to give correct results in this situation. This code is too broken to provide flow analysis. If you want extract method to work anyway, that's a different story, but you're not going to get reliable analysis to help.